File: leetcode beginners/ransomNote.py
from collection import Counter

class Solution:
    def canConstruct(self, ransomNote: str, magazine: str) -> bool:
        st1, st2 = Counter(ransomNote), Counter(magazine)
        if st1 & st2 == st1:
            return True
        return False
    
    #Counter is a class that could calculate the times a string appear and give you a dictionary
    #{"apple": 3, "banana": 2}

class Solution:
    def canConstruct(self, ransomNote: str, magazine: str) -> bool:
        chars = {}
        for char in magazine:
            chars[char] = chars.get(char, 0) + 1

        for char in ransomNote:
            if char not in chars:
                return False
            chars[char] -= 1

            if chars[char] < 0:
                return False

        return True

File: leetcode in c/ransom note.c
bool canConstruct(char* ransomNote, char* magazine) {
    
    int tmp[26] = {0};

    while (*ransomNote)
    {
        tmp[*ransomNote - 'a']++;
        ransomNote++;
    }

    while (*magazine)
    {
        tmp[*magazine - 'a']--;
        magazine++;
    }

    for (int i = 0 ; i < 26; i++)
    {
        if (tmp[i] > 0)
        {
            return false;
        }
    }
    return true;
}